src/main/resources/static/templates/UnrestrictedFileUpload/LEVEL_1/FileUpload.js

```diff
@@ -1,13 +1,46 @@
+function genericResponseHandler2(xmlHttpRequest, callBack, isJson) {
+  if (xmlHttpRequest.readyState == XMLHttpRequest.DONE) {
+    if (xmlHttpRequest.status == 200 || xmlHttpRequest.status == 401) {
+      if (isJson) {
+        callBack(JSON.parse(xmlHttpRequest.responseText));
+      } else {
+        callBack(xmlHttpRequest.responseText);
+      }
+    } else if (xmlHttpRequest.status == 400) {
+      alert("There was an error 400");
+    } else if (xmlHttpRequest.status == 500) {
+      document.getElementById("uploaded_file_info").innerHTML =
+        "No space left on the device";
+    } else {
+      alert("something else other than 200/401 was returned");
+    }
+  }
+}
+
 function addingEventListenerToUploadImage() {
   document.getElementById("upload").addEventListener("click", function () {
-    var form = document.getElementById("file");
-    var file = form.files[0];
-    var formData = new FormData();
-    formData.append("file", file);
-    let url = getUrlForVulnerabilityLevel();
-    doPostAjaxCall(uploadImage, url, true, formData);
+    const form = document.getElementById("file");
```

Comment:
I think we can allow every call from UI and let's not have a check in  the UI. thoughts?